### PR TITLE
neutron: Use internal endpoint for nova notifications

### PR DESCRIPTION
**Pull request by vuntz**
This is a new setting that got introduced in
https://review.openstack.org/#/c/291810/

Closes https://github.com/sap-oc/crowbar-openstack/issues/3

### CODE REVIEW

**Comment by vuntz** [comment]
For the record, I've tested it and it works as intended. Also, while we don't have CI here yet, the first commit already passed CI in https://github.com/crowbar/crowbar-openstack/pull/796

File: chef/cookbooks/neutron/templates/default/neutron.conf.erb [hunk 1]
```diff
@@ -849,6 +849,13 @@ pool_timeout = <%= @sql_pool_timeout %>
 # PEM encoded client certificate cert file
 # certfile =
 
+<% if node.roles.include?("neutron-server") -%>
```
Comment (tpatzig) [inline chef/cookbooks/neutron/templates/default/neutron.conf.erb:852]: and the role for the neutron-agents is anything else than neutron-server. Thats good.
Are we 100% sure, that this nova notifications are really only be sent from neutron-server and not from the agents or the computes? Is that somewhere documented?

Comment (vuntz) [reply chef/cookbooks/neutron/templates/default/neutron.conf.erb:852]: I didn't find any documentation, but the only files that seem to be using the nova notifier code are neutron/api/v2/base.py and neutron/db/db_base_plugin_v2.py, which, to my knowledge, are specific to the neutron server.

Comment (vuntz) [reply chef/cookbooks/neutron/templates/default/neutron.conf.erb:852]: Also the neutron logs seem to agree with this.

**Comment by vuntz** [comment]
(rebased as I force-pushed staging to clean things a bit with better cherry-picks)

**Review by matelakat — APPROVED** [review]

**Review by tpatzig — APPROVED** [review]

**Pull request merged**